packages/audioplayers/lib/src/audioplayer.dart

```diff
@@ -28,8 +28,10 @@ class AudioPlayer {
   PlayerState get state => _playerState;
 
   set state(PlayerState state) {
-    _playerStateController.add(state);
-    _playerState = state;
+    if (!_playerStateController.isClosed) {
+      _playerStateController.add(state);
+      _playerState = state;
+    }
```

Comment:
probably we can set the value either way, just avoid the stream when closed
```suggestion
    }
    _playerState = state;
```